test/model/unreads_test.dart

```diff
@@ -83,14 +83,17 @@ void main() {
     final Set<int> expectedMentions = {};
     for (final message in messages) {
       if (message.flags.contains(MessageFlag.read)) {
+        check(model.locatorMap).not((it) => it.containsKey(message.id));
         continue;
       }
       switch (message) {
         case StreamMessage():
+          check(model.locatorMap)[message.id].equals(TopicNarrow.ofMessage(message));
```

Comment:
These checks don't fully cover the invariant that I'd want locatorMap to maintain: it should contain exactly the messages that are in `streams` and `dms`, and agree with those data structures about where each message is. These checks appear to not cover the possibility that some messages linger in locatorMap that aren't in this `messages` argument.

I think the cleanest way to check that would be as a separate piece of logic at the end of this function. Let's build up the whole expected locatorMap, based on the other data structures; then `deepEquals` to compare.